azure/services/managedclusters/spec.go

```diff
@@ -181,6 +222,69 @@ func (s *ManagedClusterSpec) CustomHeaders() map[string]string {
 	return s.Headers
 }
 
+// buildAutoScalerProfile builds the AutoScalerProfile for the ManagedClusterProperties.
+func buildAutoScalerProfile(autoScalerProfile *AutoScalerProfile) *containerservice.ManagedClusterPropertiesAutoScalerProfile {
+	if autoScalerProfile == nil {
+		return nil
```

Comment:
If I don't set it to `nil` here, it will potentially impact anyone not managing those settings via CAPZ and create a scenario where CAPZ is continuously trying to reconcile the state. (see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2838/files#r1061928877)

Reply:
I know it's possible to express replica count for machine pool as being managed outside of CAPI/CAPZ, but it seems like the predominant assumption is that CAPZ is authoritative, where changes made externally are expected to be overwritten to match CAPZ's view.

I think following that assumption that CAPZ will always manage the autoscaler settings would be most consistent with the rest of the API, but I could see possibly adding something like the [`cluster.x-k8s.io/replicas-managed-by`](https://github.com/kubernetes-sigs/cluster-api/blob/v1.3.1/api/v1beta1/common_types.go#L134-L138) annotation which would communicate that an empty `spec.autoscalerProfile` means "don't change the current settings" (nil) instead of "assume all default values" (empty, non-nil).

Reply:
Do we want to takeover and potentially stomp on existing settings when this is released? Because of the way the settings come back from AKS I can't see a way to do this without _always_ setting all of the default values at first reconcile. That would mean it's a breaking change because we'd need to give users an opportunity to set that annotation ahead of the release, right? I would think it's more appropriate to say that if it isn't set in the manifest that it's a no-op. I might be missing something here though or over thinking this.

lmk if we should take this to a more real-time format to discuss the various scenarios I've run into with testing on this.

Reply:
@jackfrancis Have we ever run into this situation before, where the introduction of a new API field could cause CAPZ to override configuration set by the user by other means (since that was not yet exposed through CAPZ at the time)?

Reply:
SGTM